hivemind/moe/client/expert.py

```diff
@@ -1,43 +1,72 @@
-from typing import Any, Dict, Optional, Tuple
+from __future__ import annotations
+
+import os
+from concurrent.futures import Future
+from dataclasses import dataclass
+from queue import Queue
+from threading import Thread
+from typing import Any, Awaitable, Dict, Iterable, List, Optional, Sequence, Tuple, Union
 
 import torch
 import torch.nn as nn
 from torch.autograd.function import once_differentiable
 
+from hivemind import moe
```

Comment:
I agree that generic imports are bad, but this done to avoid circular import. `ConnectionHandler` is the part of `moe/server`. There is also `dht_handlers.py` which imports `moe.client.Remoteexpert`. If we try to import something from `moe/server directly` we will get circular import.


### TLDR

 - `moe.server.dht_handlers` depends on `moe.clien.expert` (already). Thus `moe.server` depends on `moe.client`
 - importing `moe.server.connection_handler` into  `moe.client.expert` creates circular deependency.